### PR TITLE
New Port Thy Sword

### PR DESCRIPTION
**Pull request by Fraxinus88**
# New Port for Thy Sword

## Game Information
- **Title**: Thy Sword
- **URL**: https://store.steampowered.com/app/560300/Thy_Sword/ OR https://gamephase.itch.io/thy-sword

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [X] AmberELEC
- [X] ArkOS
- [X] ROCKNIX
- [X] muOS

### Resolution Tests
Test all major resolutions:
- [X] 480x320
- [X] 640x480
- [X] 720x720 (RGB30)
- [X] Higher resolutions (e.g., 1280x720)

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - gameinfo.xml
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

### CODE REVIEW

**Review by Cebion — CHANGES_REQUESTED** [review]
Please review your old successful PRs for the .sh changes you did previously. If we review for example that echo is replaced by pm_message please don't use older templates without these changes. Else we'll have to request the same changes on every PR.

File: ports/thysword/Thy Sword.sh [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    pm_message "Error: Missing files in gamedata folder or game has been patched."
+fi
+
+# dos2unix in case we need it
+dos2unix "$GAMEDIR/tools/SDL_swap_gpbuttons.py"
+
+# Swap buttons
+"$GAMEDIR/tools/SDL_swap_gpbuttons.py" -i "$SDL_GAMECONTROLLERCONFIG_FILE" -o "$GAMEDIR/gamecontrollerdb_swapped.txt" -l "$GAMEDIR/SDL_swap_gpbuttons.txt"
+export SDL_GAMECONTROLLERCONFIG_FILE="$GAMEDIR/gamecontrollerdb_swapped.txt"
+export SDL_GAMECONTROLLERCONFIG="`pm_message "$SDL_GAMECONTROLLERCONFIG" | "$GAMEDIR/tools/SDL_swap_gpbuttons.py" -l "$GAMEDIR/SDL_swap_gpbuttons.txt"`"
```
Comment (cdeletre) [inline ports/thysword/Thy Sword.sh:75]:
```suggestion
export SDL_GAMECONTROLLERCONFIG="`echo "$SDL_GAMECONTROLLERCONFIG" | "$GAMEDIR/tools/SDL_swap_gpbuttons.py" -l "$GAMEDIR/SDL_swap_gpbuttons.txt"`"
```

;)

File: ports/thysword/Thy Sword.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        $ESUDO $controlfolder/xdelta3 -d -s gamedata/data.win -f ./tools/thysworditch.xdelta gamedata/game.droid && \
+        rm gamedata/data.win
+        rm -f "gamedata/place data.win here"
+	pm_message "Itch.io version of the game has been patched"
+    else
+        pm_message "Error: MD5 checksum of data.win does not match any expected version."
+        exit 1
+    fi
+else    
+    pm_message "Error: Missing files in gamedata folder or game has been patched."
```
Comment (cdeletre) [inline ports/thysword/Thy Sword.sh:66]:
```suggestion
```

I think we don't want to give an error message on each start once the game has been patched.

File: ports/thysword/Thy Sword.sh [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+# We log the execution of the script into log.txt
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+# Permissions
+$ESUDO chmod +x $GAMEDIR/tools/SDL_swap_gpbuttons.py
+$ESUDO chmod +x $GAMEDIR/gmloader
+
+# check if we have new enough version of PortMaster that contains xdelta3
+if [ ! -f "$controlfolder/xdelta3" ]; then
+  pm_message "This port requires the latest PortMaster to run, please go to https://portmaster.games/ for more info." > /dev/tty0
```
Comment (cdeletre) [inline ports/thysword/Thy Sword.sh:37]:
```suggestion
  pm_message "This port requires the latest PortMaster to run, please go to https://portmaster.games/ for more info."
```

**Review by cdeletre — CHANGES_REQUESTED** [review]

**Review by cdeletre — APPROVED** [review]

**Review by Cebion — APPROVED** [review]

**Pull request merged**